File: ALODWebUtility.CS/Printing/FormFieldParsers/FormFieldParserFactory.cs
using ALOD.Core.Interfaces;

namespace ALODWebUtility.Printing.FormFieldParsers
{
    public static class FormFieldParserFactory
    {
        public static IPDFFormFieldParser GetAdobeLiveCycleHierarchicalParser()
        {
            return new AdobeLiveCycleHierarchicalParser();
        }

        public static IPDFFormFieldParser GetNonHierarchicalParser()
        {
            return new NonHierarchicalParser();
        }

        public static IPDFFormFieldParser GetParserByName(string name)
        {
            switch (name)
            {
                case "NonHierarchicalParser":
                    return GetNonHierarchicalParser();

                case "AdobeLiveCycleHierachicalParser":
                    return GetAdobeLiveCycleHierarchicalParser();

                default:
                    return GetNonHierarchicalParser();
            }
        }
    }
}

File: ALODWebUtility/Printing/FormFieldParsers/FormFieldParserFactory.vb
Imports ALOD.Core.Interfaces

Namespace Printing.FormFieldParsers
    Public Module FormFieldParserFactory

        Public Function GetAdobeLiveCycleHierarchicalParser() As IPDFFormFieldParser
            Return New AdobeLiveCycleHierarchicalParser()
        End Function

        Public Function GetNonHierarchicalParser() As IPDFFormFieldParser
            Return New NonHierarchicalParser()
        End Function

        Public Function GetParserByName(ByVal name As String) As IPDFFormFieldParser
            Select Case name
                Case "NonHierarchicalParser"
                    Return GetNonHierarchicalParser()

                Case "AdobeLiveCycleHierachicalParser"
                    Return GetAdobeLiveCycleHierarchicalParser()

                Case Else
                    Return GetNonHierarchicalParser()
            End Select
        End Function

    End Module
End Namespace